src/transformers/tokenization_utils_base.py

```diff
@@ -1804,15 +1812,20 @@ def apply_chat_template(
             is_batched = False
 
         rendered = []
+        all_generation_indices = []
         template_kwargs = {**self.special_tokens_map, **kwargs}  # kwargs overwrite special tokens if both are present
         for chat in conversations:
             if hasattr(chat, "messages"):
                 # Indicates it's a Conversation object
                 chat = chat.messages
-            rendered_chat = compiled_template.render(
-                messages=chat, add_generation_prompt=add_generation_prompt, **template_kwargs
+            rendered_chat, generation_indices = self._render_with_assistant_indices(
+                compiled_template=compiled_template,
+                messages=chat,
+                add_generation_prompt=add_generation_prompt,
+                **template_kwargs,
             )
```

Comment:
Shouldn't we only take this path when `return_masked_labels==True`?

Reply:
yea I thought it would be cleaner to just have one function call. I changed it to go though the old path if `False` thats also fine maybe better for performance